public/Invoke-DbaDbAzSqlTip.ps1

```diff
@@ -0,0 +1,209 @@
+            if ($PSCmdlet.ShouldProcess('AzSqlTips', 'Update local cached copy of the software')) {
+                try {
+                    Save-DbaCommunitySoftware -Software AzSqlTips -LocalFile $LocalFile -EnableException
+                } catch {
+                    Stop-Function -Message 'Failed to update local cached copy' -ErrorRecord $_
+                }
+            }
+        }
+
+        # get the tips query code
+        if ($Compat100) {
+            $azTipsQuery = Get-Content (Join-DbaPath $localCachedCopy sqldb-tips 'get-sqldb-tips-compat-level-100-only.sql') -Raw
+        } else {
+            $azTipsQuery = Get-Content (Join-DbaPath $localCachedCopy sqldb-tips 'get-sqldb-tips.sql') -Raw
+        }
+
+        if ($ReturnAllTips) {
+            # if ReturnAllTips is true set the variable to 1
+            $azTipsQuery = ($azTipsQuery -replace '(?<=ReturnAllTips)(\D+)(\d+)', ('$1 1') )
+        }
+
+    }
+
+    process {
+        foreach ($instance in $SqlInstance) {
+            try {
+                Write-Message -Message ('Connecting to {0}' -f $instance)
+                $connection = Connect-DbaInstance -SqlInstance $instance -SqlCredential $SqlCredential -StatementTimeout ($StatementTimeout * 60)
+
+                if ($connection.DatabaseEngineType -ne 'SqlAzureDatabase') {
+                    Stop-Function -Message ('{0} is not an Azure SQL Database - this function only works against Azure  SQL Databases' -f $instance) -Continue
+                }
+
+                if ($AllUserDatabases) {
+                    $Database = ($connection.Databases | Where-Object name -ne 'Master').Name
+                }
+
+            } catch {
+                $failedInstConn = $true
+
+                if ($AllUserDatabases) {
+                    Write-Warning -Message ("Could not connect at instance level to {0}, so we can't get the list of databases. You'll need to specify a list of databases with -Database." -f $_)
+                    break
+                }
+
+                Write-Warning -Message ('Could not connect at instance level, so will try to connect to database. {0}' -f $_)
+
+            }
+
+            foreach ($db in $Database) {
+
+                try {
+
+                    Write-Message -Message ('Running Azure SQL Tips against {0}' -f $db)
+
+                    if ($failedInstConn) {
+                        Write-Message -Message ('Connecting to {0}.{1}' -f $instance, $db)
+                        $connection = Connect-DbaInstance -SqlInstance $instance -SqlCredential $SqlCredential -StatementTimeout $StatementTimeout -Database $db
+                    }
+
+                    Invoke-DbaQuery -SqlInstance $connection -Database $db -Query $azTipsQuery -EnableException:$EnableException | ForEach-Object {
+                        [PSCustomObject]@{
+                            ComputerName        = $connection.ComputerName
+                            InstanceName        = $connection.ServiceName
```

Comment:
The ServiceName property will always return null on Azure SQL DBs I would recommend using the Name property. That should return <Computername,Port> value.
```suggestion
                            InstanceName        = $connection.Name
```

![image](https://user-images.githubusercontent.com/11204251/217931492-457c3b37-c844-4db9-8233-0d836b26eb97.png)